actionpack/test/controller/renderer_test.rb

```diff
@@ -87,6 +87,36 @@ class RendererTest < ActiveSupport::TestCase
     assert_equal "<p>1\n<br />2</p>", render[inline: '<%= simple_format "1\n2" %>']
   end
 
+  test 'action controller metal does not have a renderer' do
```

Comment:
I think these three tests are testing implementations details. Only the tests at actionpack/test/controller/metal/renderers_test.rb are sufficient.